Develop/scripts/script.js

```diff
@@ -0,0 +1,198 @@
+$(document).ready(function () {
+  var cityArray = [];
+  var lon = "";
+  var lat = "";
+  var apiKey = "39edc21fee70b725a0dbcb6dbed85366";
+  var city = "";
+
+  retrieveCityArray();
+
+  // When user searches for city
+  $("#add-city").on("click", function (event) {
+    event.preventDefault();
+
+    storeCityArray();
+    retrieveCityArray();
+  });
+
+  // When user clicks city button
+  $(document).on("click", ".city-button", getCityClicked);
+
+  function storeCityArray() {
+    city = $("#city-input").val().trim();
+    localStorage.setItem("lastSearchedCity", JSON.stringify(city));
+
+    cityArray.push(city);
+    localStorage.setItem("cities", JSON.stringify(cityArray));
+  }
+
+  function retrieveCityArray() {
+    var retrievedArray = localStorage.getItem("cities");
+    city = JSON.parse(localStorage.getItem("lastSearchedCity"));
+
+    if (retrievedArray !== null) {
```

Comment:
Could write this like this: 

```
if (! retrievedArray) { 
  return;
}

cityArray = JSON.parse(retrievedArray);
renderCityButtons();
getLonLat();
```

Reply:
^ that is called an early return. If it's not there - don't execute the rest of the function, return early.